src/App.tsx

```diff
@@ -15,47 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum Sort {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortedBy, setSortedBy] = useState(Sort.NONE);
+  const [isVisible, setIsVisible] = useState(false);
```

Comment:
There is no need to hold whether Reset button is visible in state because it depends on values of existing states

Reply:
done